test/e2e/networking.go

```diff
@@ -101,155 +102,212 @@ var _ = Describe("Networking", func() {
-		filterNodes(nodes, func(node api.Node) bool {
-			return isNodeReadySetAsExpected(&node, true)
-		})
 
-		if len(nodes.Items) == 0 {
-			Failf("No Ready nodes found.")
-		}
-		if len(nodes.Items) == 1 {
-			// in general, the test requires two nodes. But for local development, often a one node cluster
-			// is created, for simplicity and speed. (see issue #10012). We permit one-node test
-			// only in some cases
-			if !providerIs("local") {
-				Failf(fmt.Sprintf("The test requires two Ready nodes on %s, but found just one.", testContext.Provider))
+		// Finally, we pass/fail the test based on if the container's response body, as to whether or not it was able to find peers.
+		switch {
+		case string(body) == "pass":
+			Logf("Passed.")
+			passed = true
+			return true, nil
+		case string(body) == "running":
+			Logf("test still running")
+		case string(body) == "fail":
+			if body, err = getDetails(); err != nil {
+				Failf("Failed on attempt. Error reading details: %v", err)
+				return false, err
+			} else {
+				Failf("Failed on attempt. Details:\n%s", string(body))
+				return false, nil
 			}
-			Logf("Only one ready node is detected. The test has limited scope in such setting. " +
-				"Rerun it with at least two nodes to get complete coverage.")
+		case strings.Contains(string(body), "no endpoints available"):
+			Logf("Attempt: waiting on service/endpoints")
+		default:
+			Logf("Unexpected response:\n%s", body)
 		}
-
-		podNames := LaunchNetTestPodPerNode(f, nodes, svcname, "1.6")
-
-		// Clean up the pods
-		defer func() {
-			defer GinkgoRecover()
-			By("Cleaning up the webserver pods")
-			for _, podName := range podNames {
-				if err = f.Client.Pods(f.Namespace.Name).Delete(podName, nil); err != nil {
-					Logf("Failed to delete pod %s: %v", podName, err)
-				}
-			}
-		}()
-
-		By("Waiting for the webserver pods to transition to Running state")
-		for _, podName := range podNames {
-			err = f.WaitForPodRunning(podName)
-			Expect(err).NotTo(HaveOccurred())
+		return false, nil
+	}))
+
+	if !passed {
+		if body, err := getDetails(); err != nil {
+			Failf("Timed out.  Major error : Couldn't read service details: %v", err)
+		} else {
+			Failf("Timed out. Service details :\n%s", string(body))
 		}
+	}
+	ch <- 0
+	return passed
+}
 
-		By("Waiting for connectivity to be verified")
-		passed := false
+//runNetTest Creates a single pod on each host which serves
+//on a unique port in the cluster.  It then binds a service to
+//that port, so that there are "n" nodes to balance traffic to -
+//finally, each node reaches out to ping every other node in
+//the cluster on the given port.
+//The more ports you give, the more services will be spun up,
+//i.e. one service per port.
+//To test basic pod networking, send a single port.
+//To soak test the services, we can send a range (i.e. 8000-9000).
+func runNetTest(doneTimeout Done, f *Framework, ports []int, nettestVersion string) {
```

Comment:
probably better to call this done instead of doneTimeout, since it is a channel, not a timeout value.